catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Segment.js

```diff
@@ -0,0 +1,464 @@
+                    type = slot.slotType;
+                }
+                var field = new Field.FormField({
+                    key: slot.name,
+                    name: slot.name,
+                    type: type,
+                    custom: true,
+                    isSlot: true,
+                    multiValued: type === 'string',
+                    parentId: segId
+                });
+                addValueFields(field, getSlotValue(slot, type, field.get('multiValued')));
+                field.on('change', field.validate, field);
+                array.push(field);
+            }
+        });
+    }
+
+    function getSlotValue(slot, type, multiValued) {
+        if (_.isArray(slot.value)) { //ebrim value list
+            if (multiValued) {
+                return slot.value.slice(0);
+            }if (type === 'boolean') {
+                return slot.value[0] === 'true' ? true : false;
+            }
+            return slot.value[0];
+        } else {
+            if (type === 'boolean') {
+                return slot.value === 'true' ? true : false;
+            } else if (type === 'point') {
+                return {coords: slot.value.Point.pos.split(/[ ]+/)};
+            } else {
+                return slot.value;
+            }
+        }
+    }
+
+    function generateId() {
+        counter++;
+        return 'urn:segment:id:' + new Date().getTime() + "-" + counter;
+
+    }
+
+    function getSegment(data, key) {
+        var backingData = data;
+        if (!_.isArray(data)) {
+            backingData = [data];
+        }
+        for (var index = 0; index < backingData.length; index++) {
+            var foundModel;
+
+            if (backingData[index].id === key) {
+                return backingData[index];
+            }
+
+            for (var prop in backingData[index]) {
+                if (backingData[index].hasOwnProperty(prop) && typeof backingData[index][prop] === 'object') {
+                    foundModel = getSegment(backingData[index][prop], key);
+                    if (foundModel) {
+                        return foundModel;
+                    }
+                }
+            }
+        }
+    }
+
+    var Segment = {};
+
+    Segment.Segment = Backbone.AssociatedModel.extend({
+        relations: [
+            {
+                type: Backbone.Many,
+                key: 'segments',
+                collectionType: function () {
+                    return Segment.Segments;
+                },
+                relatedModel: function () {
+                    return Segment.Segment;
+                }
+            },
+            {
+                type: Backbone.Many,
+                key: 'fields',
+                collectionType: function () {
+                    return Field.FormFields;
+                },
+                relatedModel: function () {
+                    return Field.FormField;
+                }
+            }
+        ],
+        defaults: {
```

Comment:
I can see how this could be a problem. I always set those fields so it isn't a functional problem right now. If it is ok with you I will make this update in https://codice.atlassian.net/browse/DDF-2190